### PR TITLE
Rimraf+commander

### PR DESCRIPTION
**Pull request by ghost**
@watilde 

Hi.

Quite a few changes squashed in a single PR.

My bad for not creating a separate PR for adding commander.js, but I was already working with a dirty repository and it was too late to break things apart. I hope it's OK this time. 

Now, I don't know what you think about commander.js, but it's really intuitive for parsing arguments. `npmbrew` uses what commander.js calls _subcommands_ which **requires** adding a file with the `npmbrew.<command>` call to the `bin/` directory. 

This may seem a bit unattractive at first, but in the future, adding options to subcommands will be very easy.

Besides that, I added `rimraf`, removed the files no longer needed since we are using `rimraf` and `commander`.

Let me know if there's anything strange and request specific changes and I will update this PR.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -1,7 +1,6 @@
 module.exports.clean = require("./modules/clean")
 module.exports.ls = require("./modules/ls")
 module.exports.lsRemote = require("./modules/lsRemote")
-module.exports.help = require("./modules/help")
```
Comment (watilde) [inline lib/index.js:4]: The commander is a good tool, but I think help should be a module.

File: bin/npmbrew-clean [hunk 2]
```diff
@@ -0,0 +1,2 @@
+#!/usr/bin/env node
+require(__dirname + "/../lib/").clean()
```
Comment (watilde) [inline bin/npmbrew-clean:2]: I'd like to keep bin simple. Could you please put only the following as npmbrew.js ?

``` js
require(__dirname + "/../lib/cli")
```

File: bin/npmbrew.js [hunk 3]
```diff
@@ -1,2 +1,2 @@
 #!/usr/bin/env node
-require(__dirname + '/../lib/cli')
+require(__dirname + "/../lib/cli")
```
Comment (watilde) [inline bin/npmbrew.js:2]: :+1:

**Comment by watilde** [comment]
Thanks for your nice Pull Request! I've commented some minor point. Well, could you please separate this Pull Request including Rimraf, commander?
//= including Rimraf seems easy to merge and ship.

File: lib/index.js (hunk 1 above)
Comment (ghost) [inline lib/index.js:4]: Commander.js generates help automatically. The output looks very pro to me.

File: bin/npmbrew-clean (hunk 2 above)
Comment (ghost) [inline bin/npmbrew-clean:2]: :+1:

**Comment by ghost** [comment]
Yes, it's my fault like I commented earlier. 

Sure, will do.

**Pull request closed without merging**